libs/ledger/src/consensus/stake_manager.cpp

```diff
@@ -43,26 +46,69 @@ std::size_t SafeDecrement(std::size_t value, std::size_t decrement)
   }
 }
 
+template <typename T>
+void TrimToSize(T &container, uint64_t max_allowed)
+{
+  if (container.size() >= max_allowed)
+  {
+    auto const num_to_remove = container.size() - max_allowed;
+
+    if (num_to_remove > 0)
+    {
+      auto end = container.begin();
+      std::advance(end, static_cast<std::ptrdiff_t>(num_to_remove));
+
+      container.erase(container.begin(), end);
+    }
+  }
+}
+
+template <typename T>
+void DeterministicShuffle(T &container, uint64_t random_beacon)
```

Comment:
Could we rename `random_beacon` to `entropy`, `random_number` or `seed`?

Reply:
ok